pkg/volume/flocker/plugin.go

```diff
@@ -199,9 +199,15 @@ func (b flockerBuilder) SetUpAt(dir string) error {
 		if err := b.updateDatasetPrimary(datasetID, primaryUUID); err != nil {
 			return err
 		}
+
+		newState, err := b.client.GetDatasetState(datasetID)
+		if err != nil {
+			return fmt.Errorf("The volume '%s' is not available in Flocker after migration.", dir)
+		}
+
+		b.flocker.path = newState.Path
 	}
 
```

Comment:
It appears that `b.flocker.path` is no longer set if `s.Primary == primaryUUID`. Is that intended?